src/librustc_typeck/check/mod.rs

```diff
@@ -2433,13 +2433,23 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                     if arg_count == 1 {" was"} else {"s were"}),
                 error_code);
 
-            err.span_label(sp, format!("expected {}{} parameter{}",
-                                        if variadic {"at least "} else {""},
-                                        expected_count,
-                                        if expected_count == 1 {""} else {"s"}));
             if let Some(def_s) = def_span {
                 err.span_label(def_s, "defined here");
             }
+            if let Some(sugg) = sugg {
+                let mut sugg_span = sp.end_point();
+                // remove closing `)` from the span
+                sugg_span.hi = sugg_span.lo;
+                err.span_suggestion(
+                    sugg_span,
+                    "expected one unit argument. try creating a new unit value",
```

Comment:
I confess, this message was also designed to trigger the 10 word limit for suggestions, otherwise you'd get "try creating a new unit value: `()`", which really doesn't help the target audience.

I will reword it in a shorter manner and add a way to exempt this suggestion from being shortened to a label